lib/gooddata/lcm/actions/synchronize_ldm.rb

```diff
@@ -86,11 +93,15 @@ def sync_segment_ldm(params, segment_info)
           diff_against_master = %w(diff_against_master_with_fallback diff_against_master)
             .include?(params[:synchronize_ldm].downcase)
           GoodData.logger.info "Synchronize LDM mode: '#{params[:synchronize_ldm].downcase}'"
-          if previous_master && diff_against_master
-            maql_diff_params = [:includeGrain]
-            maql_diff_params << :excludeFactRule if exclude_fact_rule
-            maql_diff_params << :includeDeprecated if include_deprecated
-            maql_diff = previous_master.maql_diff(blueprint: blueprint, params: maql_diff_params)
+          if diff_against_master
+            if previous_master
+              maql_diff_params = [:includeGrain]
+              maql_diff_params << :excludeFactRule if exclude_fact_rule
+              maql_diff_params << :includeDeprecated if include_deprecated
+              maql_diff = previous_master.maql_diff(blueprint: blueprint, params: maql_diff_params)
+            else
+              maql_diff = empty_maql_diff
```

Comment:
Could we just put this here: 
{
            "projectModelDiff" =>
              {
                "updateOperations" => [],
                "updateScripts" => []
              }
          }

Reply:
due to add more logic for maql diff LDM clients so we need to keep this to reuse